iwyu_output.cc

```diff
@@ -537,6 +538,19 @@ void IwyuFileInfo::AddForwardDeclare(const clang::NamedDecl* fwd_decl,
            << internal::GetQualifiedNameAsString(fwd_decl) << "\n";
 }
 
+void IwyuFileInfo::AddUsingDecl(const clang::UsingDecl* using_decl) {
+  CHECK_(using_decl && "using_decl unexpectedly NULL");
+  using_decl_referenced_.insert(std::make_pair(using_decl, false));
```

Comment:
I've gotten rid of the UsingDeclStatus and gone with a map<UsingDecl*, bool> instead as you suggested.  Feels cleaner overall.